### PR TITLE
Enemy waves flow with call wave button #78

### PR DESCRIPTION
**Pull request by masmartyg**
Before submitting a Pull Request please read and tick this checkbox:

- [X] I did read the [project licensing](https://github.com/crystal-bit/hacktoberfest-2020#license) and I agree with the content of this Pull Request being licensed under the same conditions.

I added the EnemyCommander scene with the sequence of the waves and the times between them.
To use it, you need to associate the WaveManager scene and the EnemyCallButton in the exported parameters.
Currently the sequence is set within the code. I tried to think about how to insert these times into EnemyWave scenes but I couldn't think of a csmart system without touching @vittorio-del-bianco 's code.
I changed the MouseFilter parameter of the Enemy scene because the hp bar took the focus on top of the EnemyCallButton.
Thanks to @vittorio-del-bianco  who, with the logic and signals set, allowed me to do a simple job.

### CODE REVIEW

File: scenes/enemy-commander/EnemyCommander.gd [hunk 1]
```diff
@@ -0,0 +1,39 @@
+tool
```
Comment (davcri) [inline scenes/enemy-commander/EnemyCommander.gd:1]: Do we need this script to be a tool? I think we can remove it but I'll wait for your confirmation about this!

File: scenes/enemy-commander/EnemyCommander.gd [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+var wave_manager
+export(NodePath) var call_button_path
+var call_button
+
+export(Array) var wave_sequence = [
+#	[time from prev. wave end to to next call button visible, time to wave spawn],
+	[2, 15],
+	[15, 30],
+]
+var wave_counter = 0
```
Comment (davcri) [inline scenes/enemy-commander/EnemyCommander.gd:14]: We have a wave counter inside the Wave Manager already, do you think we can use that one?

**Review by davcri — CHANGES_REQUESTED** [review]

File: scenes/enemy-commander/EnemyCommander.gd [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+func _ready():
+	wave_manager = get_node(wave_manager_path)
+	wave_manager.connect("last_enemy_in_actual_wave_spawned", self, "_on_last_enemy_in_actual_wave_spawned")
+	call_button = get_node(call_button_path)
+	call_button.connect("enemy_start_timeout", self, "_on_enemy_start_timeout")
```
Comment (davcri) [inline scenes/enemy-commander/EnemyCommander.gd:21]: I'll ask you also one more thing: I would like to rename the `enemy-call-button.gd` signal from `enemy_start_timeout` to `next_wave_requested`, I think it's easier to understand, what do you think?

**Comment by davcri** [comment]
Hi @masmartyg thanks for the PR, I tested it locally and it looks good to me! I just requested some small change, let me know what do you think about it!

File: scenes/enemy-commander/EnemyCommander.gd (hunk 1 above)
Comment (masmartyg) [reply scenes/enemy-commander/EnemyCommander.gd:1]: Sorry, I used it to do some test locally and forget to remove :blush:

File: scenes/enemy-commander/EnemyCommander.gd (hunk 3 above)
Comment (masmartyg) [reply scenes/enemy-commander/EnemyCommander.gd:21]: I agree with the change. I saw the review this morning on Twitch and I saw that it was difficult to understand immediately what the signal was for.

File: scenes/enemy-commander/EnemyCommander.gd (hunk 2 above)
Comment (masmartyg) [reply scenes/enemy-commander/EnemyCommander.gd:14]: You're right, we can use the wave_counter var that is inside the Wave Manager.

Note:
- Replace `wave_counter` with `wave_manager.wave_counter + 1` in lines 27..29
- Delete lines 30 and 14 (in this order)

**Comment by masmartyg** [comment]
One last thing: I don't change the TileMap, but moved on top of the other nodes for view order. This is why GitHub show the difference in the level scene.

**Review by davcri — APPROVED** [review]

**Comment by davcri** [comment]
Thanks @masmartyg for the excellent PR, I just merged it :tada: 
I had to update the `level1.tscn` to use the new path of `enemy-commander.tscn` https://github.com/crystal-bit/hacktoberfest-2020/commit/92bc4695462aa3ee2cfb190c720619a23ed5c83c but it was a quick fix!

**Pull request merged**